pkg/buildplan/filters.go

```diff
@@ -44,9 +46,12 @@ const NamespaceInternal = "internal"
 
 func FilterStateArtifacts() FilterArtifact {
 	return func(a *Artifact) bool {
-		internalIngredients := sliceutils.Filter(a.Ingredients, func(i *Ingredient) bool {
-			return i.Namespace == NamespaceInternal
-		})
+		internalIngredients := []*Ingredient{}
+		if os.Getenv(constants.InstallInternalDependenciesEnvVarName) != "true" {
+			internalIngredients = sliceutils.Filter(a.Ingredients, func(i *Ingredient) bool {
+				return i.Namespace == NamespaceInternal
+			})
+		}
```

Comment:
The checkout archive installs a `shared/zlib` package, which would be excluded by default. In order to keep things small and simple, add an env var override to avoid excluding internal packages.